sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```
@@ -131,6 +132,8 @@ private[sql] class HiveSessionCatalog(
     Try(super.lookupFunction(funcName, children)) match {
       case Success(expr) => expr
       case Failure(error) =>
+        logWarning(s"Encounter a failure during looking up function:" +
+          s" ${Utils.exceptionString(error)}")
         if (functionRegistry.functionExists(funcName)) {
```

How about `logWarning(s"Failed to look up a function: $funcName")`? I feel printing `Utils.exceptionString(error)` is a little meaningless.

---

Actually, this patch intended to print actual exception.As the description mentioned, this exception will helpful for troubleshooting.
Thanks @maropu

---

@maropu Any more suggestion?Thanks

---

cc @gatorsmile @viirya

---

Instead of logging warning here, shall we set up the cause for thrown `NoSuchFunctionException`?

---

@viirya Thanks, i will set up the cause for `NoSuchFunctionException ` later

---

I have update that set up the root cause for `NoSuchFunctionException`.